src/net.c

```diff
@@ -1331,6 +1350,14 @@ coap_read_session(coap_context_t *ctx, coap_session_t *session, coap_tick_t now)
           if (n == len) {
             size_t size = coap_pdu_parse_size(session->proto, session->read_header,
               hdr_size);
+            if (size > COAP_DEFAULT_MAX_PDU_RX_SIZE) {
+              coap_log(LOG_WARNING,
+                       "** %s: incoming PDU length too large (%lu > %u)\n",
```

Comment:
`size` is of type `size_t` hence the format string should be `%zu`.
(And as `COAP_DEFAULT_MAX_PDU_RX_SIZE` is currently defined as an `int`. Not sure if we can expect warnings here.)

Reply:
Code corrected and made `COAP_DEFAULT_MAX_PDU_RX_SIZE` an unsigned long.

Code changes pushed.

Reply:
Thank you. Regarding the format string for the second argument (`COAP_DEFAULT_MAX_PDU_RX_SIZE`) I think we should use `%lu` as `size_t` and `unsigned long` might have different sizes on some platforms (Windows? I am pretty sure I have seen this somewhere).

Reply:
Changed to %lu and pushed.